File: gcp/python/artifact-registry/artifact_registry.py
# artifact_registry.py

import pulumi
from pulumi_gcp.artifactregistry import Repository
from config import REGION, dev_repo, prod_repo
from utils import Utils

class ArtifactRegistry:

    @staticmethod
    def create_repositories(repo_list: list, prefix: str):
        repositories = {}
        for repo in repo_list:
            repo_name = Utils.resource_name(f"{prefix}-{repo}")
            repositories[repo_name] = Repository(repo_name,
                                                 location=REGION,
                                                 format="DOCKER",
                                                 repository_id=repo_name,
                                                 labels={
                                                     "createdby": "pulumi",
                                                 })
        return repositories

    @staticmethod
    def create_dev_repositories():
        return ArtifactRegistry.create_repositories(dev_repo, "dev")

    @staticmethod
    def create_prod_repositories():
        return ArtifactRegistry.create_repositories(prod_repo, "prod")

File: gcp/typescript/artifact-registry/artifactRegistry.ts
import * as pulumi from "@pulumi/pulumi";
import * as artifactregistry from "@pulumi/gcp/artifactregistry";
import { REGION, devRepo, prodRepo } from "./config";
import { Utils } from "./utils";

export class ArtifactRegistry {
    static createRepositories(repoList: string[]): { [key: string]: pulumi.CustomResource } {
        const repositories: { [key: string]: pulumi.CustomResource } = {};
        for (const repo of repoList) {
            const repoName = Utils.resourceName(repo);
            repositories[repoName] = new artifactregistry.Repository(repoName, {
                location: REGION,
                format: "DOCKER",
                repositoryId: repoName,
                labels: {
                    createdby: "pulumi",
                },
            });
        }
        return repositories;
    }

    static createDevRepositories(): { [key: string]: pulumi.CustomResource } {
        return this.createRepositories(devRepo);
    }

    static createProdRepositories(): { [key: string]: pulumi.CustomResource } {
        return this.createRepositories(prodRepo);
    }
}
